signinapp/model.py

```diff
@@ -102,6 +116,10 @@ def badges(self) -> list[BadgeAward]:
         ' The badges associated with this user '
         return BadgeAward.query.filter_by(user_id=self.id).all()
 
+    @property
```

Comment:
Does this need to be a `hybrid_property`?

Reply:
My understanding of hybrid_property is it can be used as part of SQL statements. We shouldn't ever be using this value for SQL stuff, so I don't think it should be a hybrid_property.  Though I don't know if it would be a problem if it was.

Reply:
Let's just say that my understanding was flawed and it suddenly clicked for me. All good!